File: python/infinicore/ops/exp2.py
from infinicore.lib import _infinicore
from infinicore.tensor import Tensor

def exp2(input, *, out=None):
    if out is None:
        return Tensor(_infinicore.exp2(input._underlying))
    _infinicore.exp2_(out._underlying, input._underlying)
    return out

File: src/infinicore/ops/exp2/exp2.cc
#include "infinicore/ops/exp2.hpp"

namespace infinicore::op {

common::OpDispatcher<Exp2::schema> &Exp2::dispatcher() {
    static common::OpDispatcher<Exp2::schema> dispatcher_;
    return dispatcher_;
}

void Exp2::execute(Tensor output, Tensor input) {
    dispatcher().lookup(context::getDevice().getType())(output, input);
}

Tensor exp2(Tensor input) {
    auto output = Tensor::empty(input->shape(), input->dtype(), input->device());
    exp2_(output, input);
    return output;
}

void exp2_(Tensor output, Tensor input) {
    Exp2::execute(output, input);
}

} // namespace infinicore::op